### ISSUE TITLE
Fatal: unable to open config file: Stat: 401 

### ISSUE THREAD

**Issue by Coysh** [restic/restic#4631]
Output of `restic version`
--------------------------
`restic 0.16.2 compiled with go1.21.3 on linux/amd64`

What backend/service did you use to store the repository?
---------------------------------------------------------
Backblaze

Problem description / Steps to reproduce
----------------------------------------

I'm getting a Stat: 401 error with the latest version, see below

```
root@pve:~# restic -v -r s3:s3.us-west-000.backblazeb2.com/xxx backup /dump
open repository
Fatal: unable to open config file: Stat: 401 
Is there a repository at the following location?
s3:s3.us-west-000.backblazeb2.com/xxx
```

It hangs at "open repository" for a while, then gives the error.

All access keys are fresh and confirmed working

Expected behavior
-----------------

Actual behavior
---------------

Do you have any idea what may have caused this?
-----------------------------------------------

It hasn't been updating for a while, so unsure. But it used to. The backblaze bucket is at 50gb+

Did restic help you today? Did it make you happy in any way?
------------------------------------------------------------

**Comment by MichaelEischer** [comment]
the repository URL should also include `https://` that is it should be `s3:https://s3.us-west-000.backblazeb2.com/xxx`. Did the backup work before?

If the doesn't help, then you can create a debug log by setting the environment variable `DEBUG_LOG=restic.log`.

**Comment by douglasparker** [comment]
Hey @MichaelEischer,

I'm also having this issue! I am running restic using docker, but the same error applies and it was working great before. My bucket size is 90 GB.

**Docker Run**
```docker
sudo docker run --rm \
    --hostname "$RESTIC_HOSTNAME" \
    --env RESTIC_REPOSITORY="$RESTIC_REPOSITORY" \
    --env RESTIC_PASSWORD="$RESTIC_PASSWORD" \
    --env AWS_ACCESS_KEY_ID="$RESTIC_AWS_ACCESS_KEY_ID" \
    --env AWS_SECRET_ACCESS_KEY="$RESTIC_AWS_SECRET_ACCESS_KEY" \
    --env DEBUG_LOG=/root/restic.log \
    --volume restic-cache:/root/.cache/restic \
    --volume ./restic.log:/root/restic.log \
    restic/restic:latest snapshots
```

**Environmental Variables**
```bash
RESTIC_HOSTNAME="compute-server"
RESTIC_REPOSITORY="s3:https://s3.us-west-002.backblazeb2.com/dp-backups"
RESTIC_PASSWORD="redacted"
AWS_ACCESS_KEY_ID="redacted"
AWS_SECRET_ACCESS_KEY="redacted"
```

<details>

<summary>Log</summary>

```
debug log file /root/restic.log
debug enabled
Fatal: unable to open config file: Stat: 401 
Is there a repository at the following location?
s3:https://s3.us-west-002.backblazeb2.com/dp-backups
root@compute-server:/docker/ofelia# nano restic.log
root@compute-server:/docker/ofelia# .env
Command '.env' not found, did you mean:
  command 'env' from deb coreutils (8.32-4.1ubuntu1)
Try: apt install <deb name>
root@compute-server:/docker/ofelia# nano .env
root@compute-server:/docker/ofelia# nano restic.log
root@compute-server:/docker/ofelia# cat restic.log
2024/01/30 18:46:11 restic/main.go:106  main.main       1       main []string{"/usr/bin/restic", "snapshots"}
2024/01/30 18:46:11 restic/main.go:107  main.main       1       restic 0.16.3 compiled with go1.21.6 on linux/amd64
2024/01/30 18:46:11 restic/global.go:572        main.open       1       parsing location s3:https://s3.us-west-002.backblazeb2.com/dp-backups
2024/01/30 18:46:11 restic/global.go:566        main.parseConfig        1       opening s3 repository at &s3.Config{Endpoint:"s3.us-west-002.backblazeb2.com", UseHTTP:false, KeyID:"", Secret:"", Bucket:"dp-backups", Prefix:"", Layout:"", StorageClass:"", Connections:0x5, MaxRetries:0x0, Region:"", BucketLookup:"", ListObjectsV1:false}
2024/01/30 18:46:11 s3/s3.go:42 s3.open 1       open, config s3.Config{Endpoint:"s3.us-west-002.backblazeb2.com", UseHTTP:false, KeyID:"", Secret:"", Bucket:"dp-backups", Prefix:"", Layout:"", StorageClass:"", Connections:0x5, MaxRetries:0x0, Region:"", BucketLookup:"", ListObjectsV1:false}
2024/01/30 18:46:42 s3/s3.go:87 s3.open 1       using anonymous access for "s3.us-west-002.backblazeb2.com"
2024/01/30 18:46:42 layout/layout.go:139        layout.ParseLayout      1       parse layout string "" for backend at 
2024/01/30 18:46:42 layout/layout.go:99 layout.DetectLayout     1       detect layout at 
2024/01/30 18:46:42 s3/s3.go:200        s3.(*Backend).ReadDir   1       ReadDir(keys)
2024/01/30 18:46:42 s3/s3.go:210        s3.(*Backend).ReadDir   1       using ListObjectsV1(false)
2024/01/30 18:47:13 debug/round_tripper.go:93   debug.loggingRoundTripper.RoundTrip     59      ------------  HTTP REQUEST -----------
GET /dp-backups/?location= HTTP/1.1
Host: s3.us-west-002.backblazeb2.com
User-Agent: MinIO (linux; amd64) minio-go/v7.0.66
Accept-Encoding: gzip


2024/01/30 18:47:13 debug/round_tripper.go:110  debug.loggingRoundTripper.RoundTrip     59      ------------  HTTP RESPONSE ----------
HTTP/1.1 403 
Content-Length: 180
Cache-Control: max-age=0, no-cache, no-store
Content-Type: application/xml
Date: Tue, 30 Jan 2024 18:47:13 GMT
X-Amz-Id-2: adR5uQmuobrhvNHe5blY=
X-Amz-Request-Id: 6b911863073134d3


2024/01/30 18:47:44 debug/round_tripper.go:93   debug.loggingRoundTripper.RoundTrip     59      ------------  HTTP REQUEST -----------
GET /dp-backups/?delimiter=%2F&encoding-type=url&fetch-owner=true&list-type=2&prefix=keys%2F HTTP/1.1
Host: s3.us-west-002.backblazeb2.com
User-Agent: MinIO (linux; amd64) minio-go/v7.0.66
Accept-Encoding: gzip


2024/01/30 18:47:45 debug/round_tripper.go:110  debug.loggingRoundTripper.RoundTrip     59      ------------  HTTP RESPONSE ----------
HTTP/1.1 403 
Content-Length: 180
Cache-Control: max-age=0, no-cache, no-store
Content-Type: application/xml
Date: Tue, 30 Jan 2024 18:47:44 GMT
X-Amz-Id-2: adYFul2uGbrRvE3dAbuc=
X-Amz-Request-Id: f888675569350188


2024/01/30 18:47:45 s3/s3.go:200        s3.(*Backend).ReadDir   1       ReadDir(key)
2024/01/30 18:47:45 s3/s3.go:210        s3.(*Backend).ReadDir   1       using ListObjectsV1(false)
2024/01/30 18:48:16 debug/round_tripper.go:93   debug.loggingRoundTripper.RoundTrip     117     ------------  HTTP REQUEST -----------
GET /dp-backups/?delimiter=%2F&encoding-type=url&fetch-owner=true&list-type=2&prefix=key%2F HTTP/1.1
Host: s3.us-west-002.backblazeb2.com
User-Agent: MinIO (linux; amd64) minio-go/v7.0.66
Accept-Encoding: gzip


2024/01/30 18:48:16 debug/round_tripper.go:110  debug.loggingRoundTripper.RoundTrip     117     ------------  HTTP RESPONSE ----------
HTTP/1.1 403 
Content-Length: 180
Cache-Control: max-age=0, no-cache, no-store
Content-Type: application/xml
Date: Tue, 30 Jan 2024 18:48:15 GMT
X-Amz-Id-2: adXpuNmtubg1vA3cFbl4=
X-Amz-Request-Id: 623e28ff7d17cdcd


2024/01/30 18:48:16 layout/layout.go:132        layout.DetectLayout     1       layout detection failed
2024/01/30 18:48:16 layout/layout.go:156        layout.ParseLayout      1       error: auto-detecting the filesystem layout failed, use default layout default
2024/01/30 18:48:16 layout/layout.go:139        layout.ParseLayout      1       parse layout string "default" for backend at 
2024/01/30 18:48:16 logger/log.go:52    logger.(*Backend).Stat  1       Stat(<config/>)
2024/01/30 18:48:16 sema/semaphore.go:27        sema.semaphore.GetToken 1       acquired token
2024/01/30 18:48:47 debug/round_tripper.go:93   debug.loggingRoundTripper.RoundTrip     27      ------------  HTTP REQUEST -----------
HEAD /dp-backups/config HTTP/1.1
Host: s3.us-west-002.backblazeb2.com
User-Agent: MinIO (linux; amd64) minio-go/v7.0.66


2024/01/30 18:48:47 debug/round_tripper.go:110  debug.loggingRoundTripper.RoundTrip     27      ------------  HTTP RESPONSE ----------
HTTP/1.1 401 
Content-Length: 169
Cache-Control: max-age=0, no-cache, no-store
Content-Type: application/xml
Date: Tue, 30 Jan 2024 18:48:46 GMT
Www-Authenticate: AWS4-HMAC-SHA256
X-Amz-Id-2: aZj9jRGYAYXwzhmZ0MyxmbTcZNB1moGKO
X-Amz-Request-Id: fd75e9241e8d4e7d


2024/01/30 18:48:47 logger/log.go:54    logger.(*Backend).Stat  1         stat err Stat: 401 
2024/01/30 18:48:47 restic/cleanup.go:87        main.Exit       1       exiting with status code 1
```

</details>

**Comment by douglasparker** [comment]
I found out the issue was on my end! I was using the wrong environmental variable names.

**Comment by MichaelEischer** [comment]
@douglasparker Could you edit your comment to include the environment variables that worked for you?

**Comment by douglasparker** [comment]
> @douglasparker Could you edit your comment to include the environment variables that worked for you?

I used restic via docker run in a bash script for a really long time.

I recently switched to Ofelia (a docker scheduler) and restic using docker compose.

In the bash script, I was exporting variables for later use with docker run. These variables were named differently from the variables restic uses internally.

For example:
`--env AWS_ACCESS_KEY_ID="$RESTIC_AWS_ACCESS_KEY_ID"`

When I switched to using docker compose, I started using a `.env` file for environmental variables. These variables are named as restic expects them.

For example:
`AWS_ACCESS_KEY_ID=""`

I have my own documentation (a quick reference) that I use for one off docker run commands: https://docs.douglasparker.dev/en/restic

I was basically copying the contents of the `.env` file into the terminal, but my docker run commands I was copying from my documentation were still referencing my old variable names, which weren't defined.

It was completely user error. I apologize for that!
